Assets/ZSSRichTextEditor.js

```diff
@@ -273,49 +273,66 @@ ZSSEditor.getJoinedCaretArguments = function() {
     return joinedArguments;
 };
 
-ZSSEditor.getYCaretInfo = function() {
-    var y = 0, height = 0;
+ZSSEditor.getCaretYPosition = function() {
     var sel = window.getSelection();
-    if (sel.rangeCount) {
+    // Next line is comented to prevent deselecting selection. It looks like work but if there are any issues will appear then uconmment it as well as code above.
+    //sel.collapseToStart();
+    var range = sel.getRangeAt(0);
+    var span = document.createElement('span');// something happening here preventing selection of elements
+    range.insertNode(span);
+    var topPosition = span.offsetTop;
+    span.parentNode.removeChild(span);
+    return topPosition;
+}
+
+ZSSEditor.getYCaretInfo = function() {
+    var y = 0;
+    var height = 0;
+    var selection = window.getSelection();
+    
+    if (! selection.rangeCount) { // if no selection available just return 0
+        this.caretInfo.y = y;
+        this.caretInfo.height = height;
+        return this.caretInfo;
```

Comment:
Created a [new issue](https://github.com/wordpress-mobile/WordPress-iOS-Editor/issues/529) to improve this in version 4.9.